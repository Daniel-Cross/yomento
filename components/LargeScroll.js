import React from "react";
import { ScrollView } from "react-native";
import ImageCard from "../atoms/ImageCard";
import { Container, Title } from "../constants/styles";
import { images } from "../constants/data";

const LargeScroll = () => {
  return (
    <Container w="100%" align="flex-start" p={"20px"}>
      <Title align="left">Large horizontal scroll</Title>
      <Container w="100%" direction="row">
        <ScrollView horizontal>
          {images.map((image) => (
            <ImageCard key={image} h={"200px"} w={"250px"} image={image} />
          ))}
        </ScrollView>
      </Container>
    </Container>
  );
};

export default LargeScroll;
