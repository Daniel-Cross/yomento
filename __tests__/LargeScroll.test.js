import React from "react";
import renderer from "react-test-renderer";
import LargeScroll from "../components/LargeScroll";

describe("<LargeScroll />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<LargeScroll />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
