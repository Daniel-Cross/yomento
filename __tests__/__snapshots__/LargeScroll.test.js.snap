// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LargeScroll /> renders correctly 1`] = `
<View
  align="flex-start"
  p="20px"
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
  w="100%"
>
  <Text
    align="left"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 22,
          "fontWeight": "700",
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "textAlign": "left",
        },
      ]
    }
  >
    Large horizontal scroll
  </Text>
  <View
    direction="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "100%",
        },
      ]
    }
    w="100%"
  >
    <RCTScrollView
      horizontal={true}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/i2a8PUc.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/zSA6Ndb.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/zOkbqv7.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/kfmp725.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/LfAVCG2.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/CraQCpp.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/PulYGtH.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/6xTon8Q.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/WLa8ZQb.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "space-evenly",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
                "width": "auto",
              },
            ]
          }
        >
          <Image
            h="200px"
            source={
              Object {
                "uri": "https://i.imgur.com/UOkS6t8.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 200,
                  "width": 250,
                },
              ]
            }
            w="250px"
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
